Src/tim.c

```diff
@@ -53,9 +53,9 @@ void MX_TIM16_Init(void)
   TIM_BreakDeadTimeConfigTypeDef sBreakDeadTimeConfig;
 
   htim16.Instance = TIM16;
-  htim16.Init.Prescaler = 0;
+  htim16.Init.Prescaler = 79;
   htim16.Init.CounterMode = TIM_COUNTERMODE_UP;
-  htim16.Init.Period = 0;
+  htim16.Init.Period = 11999;
```

Comment:
Did a little quick mafs, the prescaler value would give a CK_CNT (counter clock) frequency of 6 x 10^5 Hz. Converted that to seconds and multiplied it with your value for period gave a period of 0.0019 seconds, which is 1.9 ms. You should have a period of roughly 20 ms, based on this servo datasheet:
http://www.ee.ic.ac.uk/pcheung/teaching/DE1_EE/stores/sg90_datasheet.pdf
Assuming I did the math right, I think you should look for a lower prescaler value, and a higher period value.